### PR TITLE
Update SystemTextJsonToolsetVersion

### PR DESCRIPTION
**Pull request by am11**
https://github.com/dotnet/runtime/pull/107282#issuecomment-2401935510
cc @jkotas

### CODE REVIEW

**Review by jakobbotsch — APPROVED** [review]

**Review by tomeksowi — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj [hunk 1]
```diff
@@ -23,6 +23,7 @@
     <PackageReference Include="LightInject.Microsoft.DependencyInjection" Version="3.7.1" />
     <PackageReference Include="Grace.DependencyInjection.Extensions" Version="7.1.0" />
     <PackageReference Include="Stashbox.Extensions.Dependencyinjection" Version="4.2.3" />
+    <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonToolsetVersion)" />
```
Comment (am11) [inline src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:26]: One of the 3p deps here was transitively resolving 6.0.0 which was also giving the same advisory error:

> src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj(0,0): error NU1903: (NETCORE_ENGINEERING_TELEMETRY=Restore) Warning As Error: Package 'System.Text.Json' 6.0.0 has a known high severity vulnerability, https://github.com/advisories/GHSA-8g4q-xg66-9fp4

Comment (tomeksowi) [reply src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:26]: Shouldn't that be 6.0.10 as per https://github.com/dotnet/runtime/issues/108678#issuecomment-2400786310 ?

File: src/mono/wasm/Wasm.Build.Tests/Wasm.Build.Tests.csproj [hunk 2]
```diff
@@ -48,6 +48,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Playwright" Version="1.47.0" />
     <PackageReference Include="MSBuild.StructuredLogger" Version="2.2.350" />
+    <PackageReference Include="System.Text.Json" Version="$(SystemTextJsonToolsetVersion)" />
```
Comment (ilonatommy) [inline src/mono/wasm/Wasm.Build.Tests/Wasm.Build.Tests.csproj:51]: Did `WasmBuildTests` fail without it?

Comment (am11) [reply src/mono/wasm/Wasm.Build.Tests/Wasm.Build.Tests.csproj:51]: It looks like all these failures started appearing about an hour ago, which coincides with the activation of the GitHub advisory:

> **`src\mono\wasm\Wasm.Build.Tests\Wasm.Build.Tests.csproj(0,0): error NU1903: (NETCORE_ENGINEERING_TELEMETRY=Restore) Warning As Error: Package 'System.Text.Json' 6.0.0 has a known high severity vulnerability, [GHSA-8g4q-xg66-9fp4](https://github.com/advisories/GHSA-8g4q-xg66-9fp4)`**

The advisory marks this package as vulnerable, causing it to be flagged as an error during the build.

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj (hunk 1 above)
Comment (am11) [reply src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:26]: Perhaps. Lets see if we can use the centralized version (contingent of run results of this test project). This way, one less hardcoded version to worry about. 😅

**Review by sirntar — APPROVED** [review]

**Comment by am11** [comment]
The two legs github is showing as running were actually timed out on AzDO side:

> ##[error]We stopped hearing from agent NetCore-Public 99. Verify the agent machine is running and has a healthy network connection. Anything that terminates an agent process, starves it for CPU, or blocks its network access can cause this error. For more information, see: https://go.microsoft.com/fwlink/?linkid=846610

**Review by akoeplinger — APPROVED** [review]

**Comment by akoeplinger** [comment]
/ba-g failures are unrelated to the change, merging to unblock the builds.

**Pull request merged**